File: Modules/JetPack/src/main/java/com/example/william/my/jet/comparator/ArticleComparator.java
package com.example.william.my.jet.comparator;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.william.my.module.bean.ArticlesBean;

public class ArticleComparator extends DiffUtil.ItemCallback<ArticlesBean.DataBean.ArticleBean> {

    @Override
    public boolean areItemsTheSame(@NonNull ArticlesBean.DataBean.ArticleBean oldItem, @NonNull ArticlesBean.DataBean.ArticleBean newItem) {
        return oldItem.getId() == newItem.getId();
    }

    @Override
    public boolean areContentsTheSame(@NonNull ArticlesBean.DataBean.ArticleBean oldItem, @NonNull ArticlesBean.DataBean.ArticleBean newItem) {
        return oldItem.getId() == newItem.getId();
    }
}

File: Modules/Kotlin/src/main/kotlin/com/example/william/my/kotlin/comparator/ArticleComparator.kt
package com.example.william.my.kotlin.comparator

import androidx.recyclerview.widget.DiffUtil
import com.example.william.my.module.bean.ArticlesBean

class ArticleComparator : DiffUtil.ItemCallback<ArticlesBean.DataBean.ArticleBean>() {

    override fun areItemsTheSame(
        oldItem: ArticlesBean.DataBean.ArticleBean,
        newItem: ArticlesBean.DataBean.ArticleBean
    ): Boolean {
        // Id is unique.
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(
        oldItem: ArticlesBean.DataBean.ArticleBean,
        newItem: ArticlesBean.DataBean.ArticleBean
    ): Boolean {
        return newItem.id == oldItem.id
    }
}